google-cloud-trace/lib/google/cloud/trace/utils.rb

```diff
@@ -29,8 +29,22 @@ module Utils
         #
         def self.time_to_grpc time
           return nil if time.nil?
+
+          # This is called with different types of objects. Extract
+          # the nanoseconds appropriately.
+          puts "SEPH: ", time.class, time
+          nanos = case
+                  when time.is_a?(Time)
+                    time.nsec
+                  when time.is_a?(Float)
+                    # Float only appear to have 6 digits of precision. Toss the rest. :shrug:
```

Comment:
Rational is more precise than Float for keeping nano second precision when storing time as a number.

Reply:
I don't think I can effect the caller here. It's getting a float.